clang/lib/ExtractAPI/Serialization/SymbolGraphSerializer.cpp

```diff
@@ -813,52 +803,24 @@ void SymbolGraphSerializer::serializeTypedefRecord(
   Symbols.emplace_back(std::move(*Typedef));
 }
 
-Object SymbolGraphSerializer::serialize() {
-  // Serialize global variables in the API set.
-  for (const auto &GlobalVar : API.getGlobalVariables())
-    serializeGlobalVariableRecord(*GlobalVar.second);
-
-  for (const auto &GlobalFunction : API.getGlobalFunctions())
-    serializeGlobalFunctionRecord(*GlobalFunction.second);
-
-  // Serialize enum records in the API set.
-  for (const auto &Enum : API.getEnums())
-    serializeEnumRecord(*Enum.second);
-
-  // Serialize struct records in the API set.
-  for (const auto &Struct : API.getStructs())
-    serializeStructRecord(*Struct.second);
-
-  // Serialize Objective-C interface records in the API set.
-  for (const auto &ObjCInterface : API.getObjCInterfaces())
-    serializeObjCContainerRecord(*ObjCInterface.second);
-
-  // Serialize Objective-C protocol records in the API set.
-  for (const auto &ObjCProtocol : API.getObjCProtocols())
-    serializeObjCContainerRecord(*ObjCProtocol.second);
-
-  for (const auto &Macro : API.getMacros())
-    serializeMacroDefinitionRecord(*Macro.second);
-
-  for (const auto &Typedef : API.getTypedefs())
-    serializeTypedefRecord(*Typedef.second);
-
-  return serializeCurrentGraph();
+Object SymbolGraphSerializer::traverse() {
+  APISetVisitor::traverseAPISet();
```

Comment:
Function name could be better, but since the base class function implements traversal across all structures, I did not want to repeat that here by overriding to include the call to `visitCurrentGraph`.